kit/azure/buildingblocks/standard-vnet-configuration/main-backend.tf

```diff
@@ -14,35 +18,63 @@ data "azurerm_subscription" "sta_subscription" {
   subscription_id = local.sta_subscription_id
 }
 
+data "azurerm_storage_account" "tfstates" {
+  name                = local.sta_name
+  resource_group_name = local.sta_rg_name
+}
+
+# resource "azurerm_storage_container" "tfstates" {
+#   name = "tfstates-standard-vnet"
+#   storage_account_name = data.azurerm_storage_account.tfstates.name
+#   container_access_type = "blob"
+# }
+
+resource "azapi_resource" "container" {
+  type      = "Microsoft.Storage/storageAccounts/blobServices/containers@2022-09-01"
+  name      = "tfstates-standard-vnet"
+  parent_id = "${data.azurerm_storage_account.tfstates.id}/blobServices/default"
+  body = jsonencode({
+    properties = {
+      defaultEncryptionScope      = "$account-encryption-key"
+      denyEncryptionScopeOverride = true
+      immutableStorageWithVersioning = {
+        enabled = false
+      }
+      metadata     = {}
+      publicAccess = "Blob"
```

Comment:
D: As far as I understand, this means [we rely on the storage account not allowing public access to overwrite this setting](https://learn.microsoft.com/en-us/azure/storage/blobs/anonymous-read-access-configure?tabs=portal#about-anonymous-read-access) otherwise it would allow anonymous public access to terraform state stored in the blobs. If my understanding is correct, I'm in favour of not allowing public access here at all.

Reply:
good point, i've changed it to None